map_gui/src/render/traffic_signal.rs

```diff
@@ -36,14 +36,24 @@ pub fn draw_signal_stage(
             }
 
             let (yellow_light, percent) = if let Some(t) = time_left {
-                (
-                    t <= Duration::seconds(5.0),
-                    (t / stage.phase_type.simple_duration()) as f32,
-                )
+                if stage.phase_type.simple_duration() > Duration::const_seconds(0.0) {
+                    (
+                        t <= Duration::seconds(5.0),
+                        (t / stage.phase_type.simple_duration()) as f32,
+                    )
+                } else {
+                    (true, 1.0)
+                }
             } else {
                 (false, 1.0)
             };
-            let yellow = Color::YELLOW;
+            // The warning color for fixed is yellow, for anything else its orange to clue the
```

Comment:
I set a generous additional and delay time as an example, and watched the color become darker green, then jump to orange, then back to green (as the extra time was over 5s), then back to orange again, before finally going to the next stage. A little unintuitive at first, but I think it works. It's nice to be able to clearly see more detail!

Reply:
This can be changed by checking the counter. We can make it part of a second PR.